cdap-api/src/main/java/co/cask/cdap/api/DatasetConfigurer.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.module.DatasetModule;
+
+/**
+ * Provides the ability to add stream, datasets.
+ */
+public interface DatasetConfigurer {
+
+  /**
+   * Adds a {@link Stream}.
+   *
+   * @param stream {@link Stream}
+   */
+  void addStream(Stream stream);
```

Comment:
@chtyim Why do we need that? This ``addStream(Stream stream)`` will create a stream if it doesn't exist. What will ``addStream(String streamName)`` do?